crates/node/builder/src/launch/common.rs

```diff
@@ -81,9 +84,16 @@ impl LaunchContext {
 
         if !config.network.trusted_peers.is_empty() {
             info!(target: "reth::cli", "Adding trusted nodes");
-            config.network.trusted_peers.iter().for_each(|peer| {
-                toml_config.peers.trusted_nodes.insert(*peer);
-            });
+
+            // resolve trusted peers if they use a domain instead of dns
+            for peer in &config.network.trusted_peers {
+                let backoff = ConstantBuilder::default().with_max_times(config.network.dns_retries);
+                let resolved = (move || { peer.resolve() })
+                .retry(&backoff)
+                .notify(|err, _| warn!(target: "reth::cli", "Error resolving peer domain: {err}. Retrying..."))
+                .await?;
+                toml_config.peers.trusted_nodes.insert(resolved);
```

Comment:
we could extract this logic, into a separate function.

but will track this in a followup.

Reply:
Feel free to ping me for any followup if I can help